docs/configs.md

```diff
@@ -1,381 +0,0 @@
----
```

Comment:
There are a lot of existing hyperlinks to this document, and I don't think this should be removed.  Instead there can be a set of supplemental configs documents covering the shims that we can link to from this common document.  I think we would want the same for the supported_ops document, since we don't currently have plans to change what types are supported for a common operation based on which Spark version is being used.

Reply:
I wrongly thought git would forward the symlinks to the actual file. 

I have made the change now to generate the docs in two places, `docs` and `docs/SPARK_DEFAULT_VERSION`.

Reply:
But we don't want almost full copies of these docs in each shim, right?  Especially if many places are going to link to the "common" doc, how will users know which version of the doc they're supposed to reference?  Having duplicates like this is very confusing, IMO, and ideally the shims should only contain the deltas from the common document with the common document indicating Spark-version specific extensions are covered in these shim docs (ideally with hyperlinks).

Reply:
If the shims-docs contains deltas, won't that be even more confusing for the user? Like e.g. if a user is looking at the default version, he will have to go to the "version-specific" doc to see if anything is different in his version of spark. IMHO the way it is right now avoids that problem, as the user can just go to the "version-specific" doc and know exactly what execs are accelerated.

We just have to handle the linking problem. Right now we can modify the incoming links to point to the default version i.e. "3.0.1".

Reply:
> Right now we can modify the incoming links to point to the default version i.e. "3.0.1".

But then there's no indication to the user that the document they're looking at does not apply to their Spark version.

As I mentioned before, we have a core set of configs and supported operators and types, that should be a common document.  It's easy to link to and there's no redundant duplication.  Shim docs can have any extra operators and types that apply to that Spark version.  Arguably this shouldn't be shim-specific either, otherwise we get a bunch of redundant docs for ParquetCacheBatchSerializer, for example.  I think of it as a "base Spark version" thing rather than something shim-specific, but I'm curious what others think.  Personally I don't want to see 8 copies of the docs and all the churn on those files every time someone touches a typecheck or exec meta.

Reply:
This is not a simple problem.  Because of how the data is stored to generate these docs we need a version of Spark to run this against (or at least compile the code against before we generate the docs).  That means unless you log onto a data bricks server that config is not being updated. It also means that there is no good way to pick through the different parts of things and generate "deltas". We override one map entry with another in many cases, how do we indicate that in the docs? How do we store it in the original code so we can indicate it in the docs?

Reply:
>But then there's no indication to the user that the document they're looking at does not apply to their Spark version.

How is that different from today? Let's say the user is on Spark 3.1.1 today and he clicks on the link, unless I am missing something, he is taken to the default version without even mentioning that the doc is only relevant for version 3.0.1? 

This is obviously a more complicated problem than we originally thought, at least I did. Can we agree to a solution that will at least document everything. The next step can be to make it better. I will leave it to others to decide how much time we should spend on this problem. 

@revans2 I was thinking we could possibly just document the `commonExecs` instead of documenting the `exec` from GpuOverrides, that way we can only get the execs that are overridden in that specific version only for the "deltas", but again, this will be more work and I would love to do the "deltas" stuff, if it's even possible, as a follow-on. 

Thoughts?

Reply:
If we are going to rewrite things lets get a design first on what we actually want this to look like. The we can have a discussion on how to make it happen, and what limitations there are with github pages that might prevent us from doing specific things. So the first thing I would ask for you @razajafri is to come up with some kind of a proposal on how we want this to look, and how it will interact with the other doc pages. Then once we all agree on that we can start to change the code.